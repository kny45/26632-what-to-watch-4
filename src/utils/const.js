export const MovieScore = {
  BAD: 3,
  NORMAL: 5,
  GOOD: 8,
  VERRY_GOOD: 10
};

export const MovieLevel = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERRY_GOOD: `Verry good`,
  AWESOME: `Awesome`
};

export const TabName = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const GENRE_ALL = `All genres`;

export const SHOWED_ITEMS_IN_MOVIES_LIST_DEFAULT = 8;

export const TIME_OUT = 1000;

export const VideoPlayerMode = {
  SMALL_MOVIE_CARD: `small movie card`,
  BIG_MOVIE_PLAYER: `big movie player`,
};

export const AppPage = {
  MAIN_PAGE: `main screen`,
  MOVIE_PAGE: `movie screen`,
  BIG_MOVIE_PLAYER: `big movie player`,
  SIGN_IN: `sign in screen`
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  MAIN: `/`,
  SING_IN: `/sign-in`,
  MY_LIST: `/mylist`,
  FILM: `/films/:id`,
  ADD_REVIEW: `/films/:id/review`,
};
