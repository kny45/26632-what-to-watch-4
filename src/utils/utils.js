import {
  MovieScore,
  MovieLevel,
  GENRE_ALL,
} from "./const.js";

export const getMovieRatingLevel = (movieScore) => {
  if (movieScore < 0) {
    return `Rating can't be negative`;
  }
  if (movieScore < MovieScore.BAD) {
    return MovieLevel.BAD;
  }
  if (movieScore < MovieScore.NORMAL) {
    return MovieLevel.NORMAL;
  }
  if (movieScore < MovieScore.GOOD) {
    return MovieLevel.GOOD;
  }
  if (movieScore < MovieScore.VERRY_GOOD) {
    return MovieLevel.VERRY_GOOD;
  }

  return MovieLevel.AWESOME;
};

export const getTimeFromMins = (mins) => {
  const hours = Math.trunc(mins / 60);
  const minutes = mins % 60;

  return (
    hours + `h ` + minutes + `m`
  );
};

export const getFourSimilarMovies = (movie, movies) => {
  return movies
    .filter((similarMovie) =>
      similarMovie.genre === movie.genre && similarMovie.name !== movie.name)
    .slice(0, 4);
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const getMoviesByGenre = (movies, genre) => {
  return (genre === GENRE_ALL)
    ? movies
    : movies.filter((movie) => movie.genre === genre);
};

export const getGenresList = (movies) => {
  return [GENRE_ALL, ...new Set(movies.map((movie) => movie.genre))];
};

export const getShowedMovies = (movies, showedMovies) => {
  return movies.slice(0, showedMovies);
};

export const formatTime = (time) => {
  const hours = Math.floor(time / 3600);
  const minutes = Math.floor((time % 3600) / 60);
  const seconds = Math.floor(time % 60);
  return [
    hours.toString().padStart(2, `0`),
    minutes.toString().padStart(2, `0`),
    seconds.toString().padStart(2, `0`)
  ].join(`:`);
};

export const normalizeMovieData = (movie) =>
  Object.keys(movie).length
    ? {
      // : movie.id,
      name: movie.name,
      posterImage: movie.poster_image,
      previewImage: movie.preview_image,
      backgroundImage: movie.background_image,
      // : movie.background_color,
      // : movie.video_link,
      previewVideoLink: movie.preview_video_link,
      description: movie.description,
      rating: movie.rating,
      scoreCount: movie.scores_count,
      director: movie.director,
      staring: movie.starring,
      runTime: movie.run_time,
      genre: movie.genre,
      released: movie.released,
      // : movie.is_favorite,
    }
    : {};

export const normalizeMoviesData = (movies) => movies.map(normalizeMovieData);

export const normalizeCommentData = (comment) =>
  Object.keys(comment).length
    ? {
      id: comment.id,
      userId: comment.user.id,
      userName: comment.user.name,
      rating: comment.rating,
      comment: comment.comment,
      date: comment.date
    }
    : {};
