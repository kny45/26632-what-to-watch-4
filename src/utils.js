import {MovieScore, MovieLevel} from "./const.js";

export const getMovieRatingLevel = (movieScore) => {
  if (movieScore < 0) {
    return `Rating can't be negative`;
  }
  if (movieScore < MovieScore.BAD) {
    return MovieLevel.BAD;
  }
  if (movieScore < MovieScore.NORMAL) {
    return MovieLevel.NORMAL;
  }
  if (movieScore < MovieScore.GOOD) {
    return MovieLevel.GOOD;
  }
  if (movieScore < MovieScore.VERRY_GOOD) {
    return MovieLevel.VERRY_GOOD;
  }

  return MovieLevel.AWESOME;
};
