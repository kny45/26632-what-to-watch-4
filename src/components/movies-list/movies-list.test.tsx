import * as React from "react";
import * as renderer from "react-test-renderer";
import {Router} from "react-router-dom";
import MoviesList from "./movies-list";

import history from "../../history";

import {noop} from "../../utils/utils";

import {
  MoviesListType,
} from "../../types";

describe(`Render MoviesList`, () => {
  it(`Should MoviesList render correctly`, () => {
    const moviesList: MoviesListType = [
      {
        id: 1,
        name: `Fantastic Beasts: The Crimes of Grindelwald`,
        posterImage: `img/the-grand-budapest-hotel-poster.jpg`,
        backgroundColor: `#ccc`,
        backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
        videoLink: `videoLink`,
        previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
        previewVideoLink: `https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/media/video/a/e1/ae1ea61fe9c315fbd913c38d6ddc7c0d.480.mp4`,
        description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave&aposs friend and protege.`,
        rating: 8.9,
        scoreCount: 240,
        director: `Wes Andreson`,
        staring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
        runTime: 99,
        genre: `Drama`,
        released: 2014,
        isFavorite: true,
      },
      {
        id: 2,
        name: `Bohemian Rhapsody`,
        posterImage: `img/the-grand-budapest-hotel-poster.jpg`,
        backgroundColor: `#ccc`,
        backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
        videoLink: `videoLink`,
        previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
        previewVideoLink: `https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/media/video/a/e1/ae1ea61fe9c315fbd913c38d6ddc7c0d.480.mp4`,
        description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave&aposs friend and protege.`,
        rating: 8.9,
        scoreCount: 240,
        director: `Wes Andreson`,
        staring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
        runTime: 99,
        genre: `Drama`,
        released: 2014,
        isFavorite: true,
      },
      {
        id: 3,
        name: `Macbeth`,
        posterImage: `img/the-grand-budapest-hotel-poster.jpg`,
        backgroundColor: `#ccc`,
        backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
        videoLink: `videoLink`,
        previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
        previewVideoLink: `https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/media/video/a/e1/ae1ea61fe9c315fbd913c38d6ddc7c0d.480.mp4`,
        description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave&aposs friend and protege.`,
        rating: 8.9,
        scoreCount: 240,
        director: `Wes Andreson`,
        staring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
        runTime: 99,
        genre: `Drama`,
        released: 2014,
        isFavorite: true,
      },
      {
        id: 4,
        name: `Aviator`,
        posterImage: `img/the-grand-budapest-hotel-poster.jpg`,
        backgroundColor: `#ccc`,
        backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
        videoLink: `videoLink`,
        previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
        previewVideoLink: `https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/media/video/a/e1/ae1ea61fe9c315fbd913c38d6ddc7c0d.480.mp4`,
        description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave&aposs friend and protege.`,
        rating: 8.9,
        scoreCount: 240,
        director: `Wes Andreson`,
        staring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
        runTime: 99,
        genre: `Drama`,
        released: 2014,
        isFavorite: true,
      },
    ];

    const tree = renderer
    .create(
        <Router history = {history}>
          <MoviesList
            moviesList = {moviesList}
            activeItem = {null}
            onActiveItemChange = {noop}
          />
        </Router>, {
          createNodeMock: () => {
            return {};
          }
        })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
