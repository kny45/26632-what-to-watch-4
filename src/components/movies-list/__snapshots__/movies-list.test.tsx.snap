// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MoviesList Should MoviesList render correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/1"
        onClick={[Function]}
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/2"
        onClick={[Function]}
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/3"
        onClick={[Function]}
      >
        Macbeth
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/4"
        onClick={[Function]}
      >
        Aviator
      </a>
    </h3>
  </article>
</div>
`;
